src/cascadia/TerminalApp/TerminalTab.cpp

```diff
@@ -440,6 +440,31 @@ namespace winrt::TerminalApp::implementation
         control.ScrollViewport(::base::ClampAdd(currentOffset, delta));
     }
 
+    // Method Description:
+    // - Serializes the state of this tab as a series of commands that can be
+    //   executed to recreate it.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - A vector of commands
+    std::vector<ActionAndArgs> TerminalTab::BuildStartupActions() const
+    {
+        auto [args, pane] = _rootPane->BuildStartupActions();
+
+        ActionAndArgs newTabAction{};
+        newTabAction.Action(ShortcutAction::NewTab);
+        // _getNewTerminalArgs MUST be called before parsing any other options,
+        // as it might clear those options while finding the commandline
+        NewTabArgs newTabArgs{ pane->GetTerminalArgsForPane() };
+        newTabAction.Args(newTabArgs);
+
+        args.insert(args.begin(), newTabAction);
+
+        // TODO: persist focused pane? zoomed pane?

```

Comment:
The focused pane is a bit of a pain because the ids of panes are not guaranteed to be the same when recreating the tree so I can't trivially just add `FocusPaneArgs { _activePane->Id() }`. My thought was maybe having the pane add a marker FocusPane action that I can use to calculate what the id should be, remove it, then add a new one to the end?